torch_xla/csrc/tensor.cpp

```diff
@@ -125,8 +125,9 @@ XLATensor XLATensor::Create(
   return xtensor;
 }
 
-XLATensor XLATensor::Create(ir::Value ir_value, const Device& device) {
-  XLATensor xtensor(std::move(ir_value), device);
+XLATensor XLATensor::Create(ir::Value ir_value, const Device& device,
+                            xla::Shape logical_shape) {
```

Comment:
Just take the at::ScalarType

Reply:
See comment below. I was inclined to only store the logical element type, but then `XLATensor::shape()` becomes a problem.

Reply:
Done.